app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,13 @@
+module Foreman
+  class Deprecation
+    #deadline_version - is the version the deprecation is going to be deleted, the format must be a major release e.g "1.8"
+    def self.deprecation_warning(foreman_version_deadline, info)
+      raise Foreman::Exception.new(N_("Invalid version format, please enter in x.y (only major version).")) unless foreman_version_deadline.to_s.match(/\A\d[.]\d+\z/)
+      ActiveSupport::Deprecation.warn("You are using a deprecated behavior, it will be removed in version #{foreman_version_deadline}, #{info}", caller)
```

Comment:
We do, I can add a caller option to the deprecation call and then we will have who ever called the deprecated method and not the deprecation method itself.
I'm just a bit afraid that it will be a bit misleading, what do you think?